projects/user_publication/gscholar.py

```diff
@@ -0,0 +1,207 @@
+        return pub
+
+    def get_authors(self, pub):
+        """returns a list of authors in a publication
+
+        Args:
+            pub (dict): scholarly return of the publication
+
+        Returns:
+            list: list of authors in ["firstname lastname",] format
+        """
+        bibt = pub['bibtex']
+        bibt = utils.decode_unicode_text(bibt)
+        match = re.search(r'author\s*=\s*\{([\w\s-,]+)\}', bibt)
+        if match:
+            authors_t = match.groups()[0]
+        else:
+            # this mostly never happens - if it does - log it
+            logger.info(f"cannot parse bibtex, so ignoring - {bibt}")
+            return
+        if "and" in authors_t:
+            authors = authors_t.split(" and ")
+        else:
+            authors = [authors_t]
+        parsed_authors = []
+        for author in authors:
+            parsed_authors.append(utils.parse_author(author))
+        return authors
+
+    def get_pub_model(self, pub):
+        return Publication(
+            title=utils.decode_unicode_text(pub["bib"]["title"]),
+            year=pub["bib"]["pub_year"],
+            author=" and ".join(self.get_authors(pub)),
+            entry_created_date=datetime.date.today(),
+            publication_type=utils.get_pub_type(pub["bib"]["ENTRYTYPE"]),
+            bibtex_source="{}",
+            added_by_username="admin",
+            forum=pub["bib"]["booktitle"],
+            link=pub["pub_url"],
+            source="gscholar",
+            status=Publication.STATUS_IMPORTED,
+        )
+
+
+def pub_import(
+    dry_run=True, year_low=2014, year_high=datetime.now().year
+):
+    """Returns Publication models of all publicationls that ref chameleon
+    and are not in database already
+    - Checks if there is a project associated with the publication
+
+    Args:
+        dry_run (bool, optional): Defaults to True.
+        year_low (int, optional): Defaults to 2014.
+        year_high (int, optional): Defaults to datetime.now().year.
+    """
+    gscholar = GoogleScholarHandler()
+    pubs = []
+    for chameleon_pub in ChameleonPublication.objects.filter(id=14):
+        ch_pub = gscholar.get_one_pub(chameleon_pub.title)
+        cited_pubs = gscholar.get_cites(ch_pub, year_low=year_low, year_high=year_high)
+        for cited_pub in cited_pubs:
+            authors = gscholar.get_authors(cited_pub)
+            cited_pub_title = cited_pub['bib']['title']
+            if not authors:
+                continue
+            proj = utils.guess_project_for_publication(
+                authors, cited_pub["bib"]["pub_year"]
+            )
+            if not proj:
+                continue
+            if ChameleonPublication.objects.filter(title__iexact=cited_pub_title).exists():
+                logger.info(f"{cited_pub_title} is a chameleon publication - ignoring")
+                continue
+            if Publication.objects.filter(title=cited_pub_title, project_id=proj).exists():

```

Comment:
Same thing here re: Mark's comment about what constitutes a duplicate.

Reply:
as included in Mark's comment I will talk to Kate and we can decide on a good strategy for this.
Will include that change in the next PR - google citations branch